apps/trading/lib/hooks/use-current-programs.ts

```diff
@@ -0,0 +1,289 @@
+
+    volumeDiscountProgram = {
+      benefitTiers,
+      details,
+    };
+  }
+
+  return {
+    referralProgram,
+    volumeDiscountProgram,
+    loading,
+    error,
+  };
+};
+
+function prepareReferralBenefitTiers(
+  program: ReferralProgramFieldsFragment,
+  fees: Fees
+): ReferralBenefitTier[] {
+  const benefitTiers = program.benefitTiers.map((t, i) => {
+    const benefitTier: ReferralBenefitTier = {
+      tier: i + 1, // sorted in asc order, hence first is the lowest tier
+      rewardFactors: parseFactors(t.referralRewardFactors),
+      rewardFactor: calcRewardFactor(
+        fees,
+        parseFactors(t.referralRewardFactors)
+      ),
+      discountFactors: parseFactors(t.referralDiscountFactors),
+      discountFactor: calcDiscountFactor(
+        fees,
+        parseFactors(t.referralDiscountFactors)
+      ),
+      minimumRunningNotionalTakerVolume: Number(
+        t.minimumRunningNotionalTakerVolume
+      ),
+      epochs: Number(t.minimumEpochs),
+    };
+    return benefitTier;
+  });
+
+  return benefitTiers;
+}
+
+function prepareReferralStakingTiers(
+  program: ReferralProgramFieldsFragment
+): ReferralStakingTier[] {
+  const stakingTiers = sortBy(program.stakingTiers, (t) =>
+    parseFloat(t.referralRewardMultiplier)
+  ).map((t, i) => {
+    return {
+      tier: i + 1,
+      ...t,
+    };
+  });
+
+  return stakingTiers;
+}
+
+function prepareVolumeDiscountBenefitTiers(
+  program: VolumeDiscountFieldsFragment,
+  fees: Fees
+): VolumeDiscountBenefitTier[] {
+  const benefitTiers = program.benefitTiers.map((t, i) => {
+    return {
+      tier: i + 1,
+      discountFactors: parseFactors(t.volumeDiscountFactors),
+      discountFactor: calcDiscountFactor(
+        fees,
+        parseFactors(t.volumeDiscountFactors)
+      ),
+      minimumRunningNotionalTakerVolume: Number(
+        t.minimumRunningNotionalTakerVolume
+      ),
+    };
+  });
+
+  return benefitTiers;
+}
+
+function getFees(data: CurrentProgramsQuery | undefined): Fees {
+  const buybackFee = data?.defaultBuybackFee?.value
+    ? Number(data.defaultBuybackFee.value)
+    : 0;
+  const infrastructureFee = data?.defaultInfrastructureFee?.value
+    ? Number(data.defaultInfrastructureFee.value)
+    : 0;
+  const makerFee = data?.defaultMakerFee?.value
+    ? Number(data.defaultMakerFee.value)
+    : 0;
+  const treasuryFee = data?.defaultTreasuryFee?.value
+    ? Number(data.defaultTreasuryFee.value)
+    : 0;
+  const feesFromActiveMarkets = removePaginationWrapper(
+    data?.feesPerMarket?.edges
+  )
+    .filter((m) => isMarketActive(m.state))
+    .map((m) => m.fees.factors);
+  const liquidityFees = feesFromActiveMarkets.map((f) =>
+    Number(f.liquidityFee)
+  );
+  return {
+    buybackFee,
+    infrastructureFee,
+    liquidityFee: min(liquidityFees) || 0,
+    makerFee,
+    treasuryFee,
+  };
+}
+
+function calcRewardFactor(fees: Fees, factors: Factors) {
+  const MF = BigNumber(fees.makerFee).times(factors.makerFactor);
```

Comment:
Coming to this code fresh, its not immediately clear what these variables are, and all caps should ideally be reserved for true constants. Can we just name these `makerFee` etc